### ISSUE TITLE
Google Spreadsheet API requires Authentication Credentials

### ISSUE THREAD

**Issue by sm86** [dbpedia/community-viewer#1]
I found out that getting data from Google Sheets does require an Authentication Credentials. 
The details about accessing published data looks ambiguous.  
So do we have some public credentials to use from dbpedia community? Or should we think of some other solutions like scarping? Else, can it be stored as some private csv file? 
Ref: https://www.npmjs.com/package/google-spreadsheet#unauthenticated-access-read-only-access-on-public-docs
https://developers.google.com/sheets/api/quickstart/js

**Comment by jimkont** [comment]
The gdoc can be set to have view permissions for the public (anyone with the link can view the doc)

**Comment by sm86** [comment]
Yes, @jimkont there are permissions for everyone to view the file. 
> ``If you wish to work with a Google Spreadsheet without authenticating, not only must the Spreadsheet in question be visible to the web, but it must also have been explicitly published using "File > Publish to the web" menu option in the google spreadsheets GUI. Many seemingly "public" sheets have not also been "published" so this may cause some confusion.
Ref: https://www.npmjs.com/package/google-spreadsheet#unauthenticated-access-read-only-access-on-public-docs
The Sheet stated in idea tested using:
https://spreadsheets.google.com/feeds/list/1SaoJEcpi8v240TVrjT9HW9BwVnvCkq3YTcGqXlpB-Z8/od6/public/values?alt=json

To publish a file, open Doc, simply go to File->Publish to Web.

I have also found a library which can help me without above step but with few functionality. (http://chriszarate.github.io/sheetrock/). But I think publishing would be a better option.

**Comment by jimkont** [comment]
We will only need to have control on editing the file so having `publish on the web` enabled is fine.

**Comment by sm86** [comment]
Thanks for reply. The owner of the gdocs has to enable "publish to web" option for it.
Steps to publish: Just open Doc, simply go to File->Publish to Web.

**Comment by parthsuresh** [comment]
@sm86, Sheetrock.js wouldn't be a good choice here as it does not support formatted text, including hyperlinks.

> Plain text : Sheetrock doesn’t handle formatted text. Any formatting you’ve applied to your data—including hyperlinks—probably won’t show up.
Ref : http://chriszarate.github.io/sheetrock/

I feel that displaying the hyperlinks as plain text wouldn't be very user-friendly.

**Comment by sm86** [comment]
@parthsuresh I felt the same, but the sheet isn't published yet. I am waiting for @jimkont to publish to web.

**Comment by jimkont** [comment]
sorry, we are flexible with the GSheet structure, feel free to use one of your own (similar to [this](https://docs.google.com/spreadsheets/d/1SaoJEcpi8v240TVrjT9HW9BwVnvCkq3YTcGqXlpB-Z8/edit#gid=0)) and for now use the columns: name, homepage, profile_image, affiliation, role.
Cells with multiple values are separated with ';'

I don't think that we need formatted text atm and we can add a few more columns easily if the rest are in place

Thanks!

**Comment by sm86** [comment]
Thanks @jimkont . Started working on that 👍
